docs/debugging.md

```diff
@@ -25,13 +25,50 @@ it's inefficient to wade through it when looking for the cause of a failure.
 
 ## Artifacts
 While command outputs as viewed in blue ocean or the console are useful, they
-don't tell the whole story. Jobs can produce artifacts which are stored
-for a [set time](https://github.com/rcbops/rpc-gating/blob/master/playbooks/upload_to_swift.yml) after the completion of a job. Any job can produce artifacts
-and use the provided functions to publish those. Jobs artifacts should include
-everything required to debug a build.
+don't tell the whole story. Jobs can be configured to publish artifacts to a
+Rackspace Cloud Files container, which automatically has CDN and web browsing
+enabled.
+
+Artifact gathering and publishing is configured by creating a file named
+``component_metadata.yml`` in the root of the git repository which has the
+following format:
+
+``` yaml
+"artifacts": [
+  {
+      "type": "log",
+      "source": "/foo",
+  },
+  {
+      "type": "file",
+      "source": "/bar"
+  },
+  {
+      "type": "file",
+      "source": "/baz",
+      "dest": "fuzz",
+      "expire_after": "300",
+  }
+]
+```
+
+The above configuration will upload ``log`` files from every job, sourced from
+the ``/foo`` directory. These will be published and set to expire after 30 days.
+The link to where they are published will be available and displayed
+prominently in the Jenkins job result.
 
-To view artifacts, click "Build Artifact Links" in the sidebar from the standard
-Jenkins build page.
+If will also publish two sets of ``file`` artifacts, which are intended to be
+deployment artifacts. The ``source`` argument can be a file or folder which
+will be uploaded recursively. If a special destination subdirectory in the
+container is desired, the ``dest`` argument can be used to define the relative
+path from the root of the container where the artifacts should be published
+into. The ``expire_after`` argument can also optionally be used to set the
+number of seconds until the artifact expires and will automatically be removed.
+
+To view ``file`` artifacts, navigate to ``<repo name>.re.rpc.rackspace.com``
+(eg: http://rpc-octavia.re.rpc.rackspace.com). If this is not available, please
+contact the RE team to create the convenience alias in DNS for the git
```

Comment:
Is there a procedure for this somewhere?

Reply:
Not currently. I will be adding it to confluence once this is accepted, and once I have a sample repo to figure out the procedure.